dkg-gadget/src/debug_logger.rs

```diff
@@ -0,0 +1,60 @@
+use dkg_logging::{debug, error, info, trace, warn};
+use std::{io::Write, sync::Arc};
+
+#[derive(Clone, Debug)]
+pub struct DebugLogger {
+	identifier: Arc<String>,
+	to_file_io: Option<tokio::sync::mpsc::UnboundedSender<String>>,
+}
+
+impl DebugLogger {
+	pub fn new<T: ToString>(identifier: T, file: Option<std::fs::File>) -> Self {
+		// use a channel for sending file I/O requests to a dedicated thread to avoid blocking the
+		// DKG workers
+		if let Some(mut file) = file {
+			let (tx, mut rx) = tokio::sync::mpsc::unbounded_channel();
+
+			tokio::task::spawn(async move {
+				while let Some(message) = rx.recv().await {
+					writeln!(file, "{message:?}").unwrap();
+				}
+			});
+
+			Self { identifier: Arc::new(identifier.to_string()), to_file_io: Some(tx) }
+		} else {
+			Self { identifier: Arc::new(identifier.to_string()), to_file_io: None }
+		}
+	}
+
+	pub fn trace<T: std::fmt::Debug>(&self, message: T) {
+		self.log_to_file("trace", &message);
+		trace!(target: "dkg_gadget", "[{}]: {message:?}", self.identifier);
```

Comment:
I believe you've removed all file identifiers in place for `dkg_gadget`, which removes a lot of context about our logs, no? We now won't support logs from:
- `dkg_gadget::async_protocol::keygen`
- `dkg_gadget::async_protocol::signing`
etc.

Reply:
Added relevant functions to allow debugging to the different targets